### PR TITLE
(139) Send SOR codes to UH

### PR DESCRIPTION
**Pull request by dgmstuart**
if we have an SOR code, send it to UH, if not send just the repair request parts

To test, add a valid SOR code into a leaf node of the yaml file

### CODE REVIEW

File: app/services/create_repair.rb [hunk 1]
```diff
@@ -4,22 +4,33 @@ def initialize(api: HackneyApi.new)
...
+    }.tap do |hash|
+      hash[:repairOrders] = create_work_order_params(params) if params.sor_code
+    end
+  end
+
+  def create_work_order_params(params)
+    [
+      {
+        jobCode: params.sor_code,
+        propertyReference: params.property_reference,
```
Comment (leeky) [inline app/services/create_repair.rb:28]: I thought we were going to change the API endpoint to remove the duplication of `propertyReference` ?

Also, it's referred to as both `propertyRef` and `propertyReference` in the same hash, which feels odd.

File: app/services/create_repair.rb [hunk 2]
```diff
@@ -35,6 +46,10 @@ def property_reference
     def priority
       'N'
     end
+
+    def sor_code
+      @answers.fetch('diagnosis', {})['sor_code']
```
Comment (leeky) [inline app/services/create_repair.rb:54]: Should this be `@answers.fetch('diagnosis', {}).fetch('sor_code')`, so avoid a `NilClass` error if `diagnosis` is nil..?

File: app/models/question.rb [hunk 3]
```diff
@@ -13,6 +15,12 @@ def answers_for_collection
...
   end
 
+  def answer_data(chosen_answer)
+    answers.detect { |answer| answer['text'] == chosen_answer }.tap do |data|
+      raise InvalidAnswerError if data.nil?
+    end
+  end
+
   def redirect_path_for_answer(chosen_answer)
     answer_hash = answers.detect { |answer| answer['text'] == chosen_answer }
```
Comment (leeky) [inline app/models/question.rb]: If it worth using the  new `answer_data` method here?

**Review by leeky — CHANGES_REQUESTED** [review]

File: app/models/question.rb (hunk 3 above)
Comment (leeky) [reply app/models/question.rb]: As discussed, you did this in the next commit!

File: app/services/create_repair.rb (hunk 2 above)
Comment (dgmstuart) [reply app/services/create_repair.rb:54]: @leeky in what circumstance could we get a NilClass error?
I think all the possibilities are covered by tests - here are the specific cases:

```
> @answers = {}
> @answers.fetch('diagnosis', {})['sor_code']
=> nil
                              
> @answers = { 'diagnosis' => { 'next' => 'foo' }}
> @answers.fetch('diagnosis', {})['sor_code']
=> nil

> @answers = { 'diagnosis' => { 'sor_code' => 'foo' }}
> @answers.fetch('diagnosis', {})['sor_code']
=> "foo"
```

File: app/services/create_repair.rb (hunk 1 above)
Comment (dgmstuart) [reply app/services/create_repair.rb:28]: Totally agree - I should have mentioned in the commit message: these values work with the Hackney API as it is currently deployed. When https://trello.com/c/txl3rMJK/132-apply-structure-to-endpoints is done we can change it to something more sensible.

File: app/services/create_repair.rb (hunk 2 above)
Comment (dgmstuart) [reply app/services/create_repair.rb:54]: For completeness:

```
> @answers = { 'diagnosis' => { 'next' => 'foo' }}
> @answers.fetch('diagnosis', {}).fetch('sor_code')
KeyError: key not found: "sor_code"
```

File: app/services/create_repair.rb (hunk 1 above)
Comment (leeky) [reply app/services/create_repair.rb:28]: 👍

**Review by leeky — APPROVED** [review]

**Comment by leeky** [comment]
LGTM! :shipit:

File: app/services/create_repair.rb (hunk 2 above)
Comment (leeky) [reply app/services/create_repair.rb:54]: My mistake - I'd forgotten that requesting the value of a missing key in a `Hash` returns `nil`.

**Pull request merged**